mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/process/VoidCompletionCallback.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.process;
+
+import org.mule.runtime.api.meta.model.ComponentModel;
+
+/**
+ * This callback is how a {@link ComponentModel components} notifies that its execution is completed,
```

Comment:
the consumers of this callback don't know anything about component model. Rephrase refering to the actual case, and ideally even add an example.